packages/instrumenter/src/transformers/mutant-collector.ts

```diff
@@ -16,9 +16,13 @@ export class MutantCollector {
    * @param mutationSpecs the named node mutation to be added
    * @returns The mutant (for testability)
    */
-  public add(fileName: string, mutationSpecs: NamedNodeMutation): Mutant {
+  public add(fileName: string, mutationSpecs: NamedNodeMutation, lineOffset = 0): Mutant {
     mutationSpecs.replacement.end = mutationSpecs.original.end;
     mutationSpecs.replacement.start = mutationSpecs.original.start;
+    if (mutationSpecs.original.loc) {
+      mutationSpecs.original.loc.start.line += lineOffset;
```

Comment:
Wait... we're still mutating the original node here, right? So this is still the same issue, no?  It's probably better to push this `lineOffset` as a property on the `Mutant` ("packages/instrumenter/src/mutant.ts") class and use that value in `toApiMutant` to offset the location.

Also, the `Mutant` has a `range` property. It basically is used as another way to identify the mutant location based on the character index in the file. That also needs to be offset correctly. For example, the `TypeScriptChecker` plugin uses range to place mutants 1-by-1 in the code to validate them on compile errors. If we want to keep supporting `.vue` files with a `<script lang="ts">`, we will need to offset the range correctly as well.

Reply:
About first part of your message - looks like it works because we mutate exact node, not all

Reply:
Thanks, this looks great.

> About first part of your message - looks like it works because we mutate exact node, not all

Yes, but if 1 node is mutated twice (by 2 mutators for example), the offset would still be applied twice I think.